helpers/app.rb

```diff
@@ -0,0 +1,123 @@
+  attr_reader :authors, :genres, :labels, :sources, :items
+
+  def initialize
+    @authors = File.exist?(AUTHORS_PATH) ? authors_to_array(AUTHORS_PATH) : []
+    @genres = File.exist?(GENRES_PATH) ? genres_to_array(GENRES_PATH) : []
+    @labels = File.exist?(LABELS_PATH) ? labels_to_array(LABELS_PATH) : []
+    @sources = File.exist?(SOURCES_PATH) ? sources_to_array(SOURCES_PATH) : []
+    @items = File.exist?(ITEMS_PATH) ? items_to_array(ITEMS_PATH) : []
+    create_files_folders
+    run
+  end
+
+  def add_author
+    first_name = get_text('Please enter author\'s first name: ')
+    last_name = get_text('Please enter author\'s last name: ')
+    authors << Author.new(first_name, last_name)
+    puts 'The author was successfully added!'
+    puts
+  end
+
+  def add_genre
+    name = get_text('Please enter genre\'s name: ')
+    genres << Genre.new(name)
+    puts 'The genre was successfully added!'
+    puts
+  end
+
+  def add_label
+    name = get_text('Please enter label\'s name: ')
+    labels << Label.new(name)
+    puts 'The label was successfully added!'
+    puts
+  end
+
+  def add_source
+    name = get_text('Please enter source\'s name: ')
+    sources << Source.new(name)
+    puts 'The source was successfully added!'
+    puts
+  end
+
+  def add_item
+    message = []
+    message << 'The author list is empty!' if authors.empty?
+    message << 'The genre list is empty!' if genres.empty?
+    message << 'The label list is empty!' if labels.empty?
+    message << 'The source list is empty!' if sources.empty?
+
+    if message.empty?
+      author_index = get_array_index(authors, 'List of authors: ', 'Please select the number of an author: ')
+      genre_index = get_array_index(genres, 'List of genres: ', 'Please select the number of a genre: ')
+      label_index = get_array_index(labels, 'List of labels: ', 'Please select the number of a label: ')
+      source_index = get_array_index(sources, 'List of sources: ', 'Please select the number of a source: ')
+
+      option = choose_item_option
+
+      item = create_item(option)
+      authors[author_index].add_item(item)
+      genres[genre_index].add_item(item)
+      labels[label_index].add_item(item)
+      sources[source_index].add_item(item)
+
+      items << item
+      puts 'Item was add successfully'
+      puts
+    else
+      display_array(message, "New item can't be added because: ")
+    end
+  end
+
+  def run
+    loop do
+      display_options
+      number = gets.chomp.to_i
+      case number
+      when 1 then add_author
+      when 2 then add_genre
+      when 3 then add_label
+      when 4 then add_source
+      when 5 then add_item
+      when 6 then display_array(authors, 'List of authors: ')
+      when 7 then display_array(genres, 'List of genres: ')
+      when 8 then display_array(labels, 'List of labels: ')
+      when 9 then display_array(sources, 'List of sources: ')
+      when 10 then display_array(items, 'List of items: ')
+      when 0
```

Comment:
- [ ] In this list you are missing options for adding
- Books
- Music album
- Game
- List all books
- List all music albums
- List of games
Please address that
<img width="1178" alt="Screenshot 2023-09-14 at 19 08 38" src="https://github.com/M-Anwar-Hussaini/Catalog-Of-Things/assets/44745855/4ba235aa-1710-4434-a23f-61301c2e3da2">

Reply:
Hi @DeliceLydia.
We have handled what you are saying in another way.
- [x] The number menu item contains adding all 4 items such as `Book`, `Music Album`, `Game`, and `Move` if there is at least one item for `Genre`, `Source`, `Author` and `Label`, It is a sub-menu of the 5th menu option.
- [x] The number 10 item displays all items from the 4 subclasses of `Items` such as  `Book`, `Music Album`, `Game`.

- Select to add one item:
![image](https://github.com/M-Anwar-Hussaini/Catalog-Of-Things/assets/131291114/8aefdc1f-77a1-4837-80a0-a9a9b283f7b2)

- Add a book:
![image](https://github.com/M-Anwar-Hussaini/Catalog-Of-Things/assets/131291114/dd519974-30f9-4b15-95c9-eff3d01dffef)

- Add a game:
![image](https://github.com/M-Anwar-Hussaini/Catalog-Of-Things/assets/131291114/5f5b9710-d0f7-4631-b5a9-0682937adab8)

- Add a movie:
![image](https://github.com/M-Anwar-Hussaini/Catalog-Of-Things/assets/131291114/2cac28b3-f73f-4057-9c5b-dae0618d1878)

- Add a music album
![image](https://github.com/M-Anwar-Hussaini/Catalog-Of-Things/assets/131291114/0cc89f66-205a-460b-81b9-2c4139a63d92)

- List all items:
![image](https://github.com/M-Anwar-Hussaini/Catalog-Of-Things/assets/131291114/ee785ae2-76ba-4eb7-9e84-d8b637814204)